tests/loops/test_loops.py

```diff
@@ -492,10 +493,10 @@ def configure_optimizers_multiple(self):
 
     # need to remove these elements for comparison; comparing with `fit_loop.state_dict()` would require the
     # fit loop to have an iterator, which is only available during training
-    checkpoint["loops"]["fit_loop"]["state_dict"]["dataloader_state_dict"] = ANY
-    assert state_dict == checkpoint["loops"]["fit_loop"]
-
```

Comment:
Why is this moved?

Reply:
Because `dataloader_state_dict` is optional. And as you override it and reload it, it would break as  dataloader_state_dict isn't None by ANY.

Reply:
The tests seem to pass for me when these changes are undone.

And if what you say is still a problem, the key can be dropped after the assertion